MediaContentSynchronization/Model/BatchGenerator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronization\Model;
+
+class BatchGenerator
```

Comment:
Can we use the collections/search request pagination to iterate entities instead of loading them all and then passing through a batch generator?

Reply:
i will recheck the case, thanks :+1: